### ISSUE TITLE
Batch normalization support

### ISSUE THREAD

**Issue by jdh8** [tenstorrent/tt-metal#12253]
To compile `aten._native_batch_norm_legit*`, we need a batch norm op in `tt-metal`.

1. Training mode requires native support (to update parameters on the fly)
2. Probable performance gain in inference mode, compared with a handful of pointwise ops (`add`, `mul`, `rsqrt`)

**Comment by mrakitaTT** [comment]
Hi @eyonland, just to check is this already on your radar and is somebody working on it?

We can decompose batchnorm in compiler into smaller ops, but we would prefer to avoid that work if ttnn will soon have native support for batchnorm.

**Comment by eyonland** [comment]
@tt-aho  and @yugaoTT , would either of you be able to look into this?

**Comment by mrakitaTT** [comment]
> we will be taking on this work

Great, thanks for the update @eyonland!

> Is the intent here to help with improving performance? If so, how much of an improvement do you believe this to be for your training?

There are couple of intents from our side:
- To avoid workarounds in compiler - If we can just lower batch norm into `ttnn.batch_norm` then we don't have to implement decomposition workaround to decompose batch norm into bunch of smaller ops that compositely make batch norm. This is our main concern.
- Performance - We hope that it would be more performant if there is dedicated ttnn op for batch norm than manually calling bunch of ops in pipeline to produce the same result, but this is not our main concern at the moment. I am also aware that ttnn could still in the background do the same thing and do the decompose.

**Comment by eyonland** [comment]
This needs to be implemented as device op.

**Comment by VirdhatchaniKN** [comment]
Training mode support is provided https://github.com/tenstorrent/tt-metal/pull/16432 and will be merged in a day.
Inference mode support is in progress

**Comment by VirdhatchaniKN** [comment]
> Training mode support is provided [#16432](https://github.com/tenstorrent/tt-metal/pull/16432) and will be merged in a day. Inference mode support is in progress

Inference mode support merged : https://github.com/tenstorrent/tt-metal/pull/16592
Working on providing fp32 support

**Comment by VirdhatchaniKN** [comment]
Fp32 support for inference mode : https://github.com/tenstorrent/tt-metal/pull/17587
Working on Fp32 for training mode

**Comment by VirdhatchaniKN** [comment]
Closing issue - tracking Fp32 support here : https://github.com/tenstorrent/tt-metal/issues/17768

**Comment by VirdhatchaniKN** [comment]
fp32 support completed and merged
